Automattic/simplenote-ios
# Notes List: Swifting UITableViewDelegate

### Details:
In this PR we're reimplementing UITableViewDelegate conformance in Swift

Since we're now gonna be dealing with entities of multiple kinds (Tags, History items), Swift is gonna help us make things look much more elegant.

Ref. #507 
cc @bummytime (Thanks in advance!!)

### Test
1. Log into your account
2. Verify your notes list loads properly
3. Verify all of the actions that show up whenever you Swipe a note (Pin / Unpin / Share / Delete)
4. Verify all of the actions that show up when you Swipe over a deleted note (Restore / Trash)

Please repeat for iPad devices

### Release
These changes do not require release notes.
